src/opa.ts

```diff
@@ -253,19 +253,22 @@ function getOpaPath(context: vscode.ExtensionContext | undefined, path: string,
             'open-policy-agent/opa',
             'OPA is either not installed or is missing from your path, would you like to install it?',
             (release: any) => {
-                // release.assets.name contains {'darwin', 'linux', 'windows'}
+                // release.assets.name contains {'darwin', 'linux', 'windows'} and {'arm64', 'amd64'}
+                // in the format `opa_$OS_$ARCH...` so we can use that to search for the appropriate asset
                 const assets = release.assets || [];
                 const os = process.platform;
+                // node can return many values for process.arch but OPA only supports arm64 and amd64
+                const arch = process.arch.indexOf('arm') !== -1 ? 'arm64' : 'amd64';
                 let targetAsset: { browser_download_url: string };
                 switch (os) {
                     case 'darwin':
-                        targetAsset = assets.filter((asset: { name: string }) => asset.name.indexOf('darwin') !== -1)[0];
+                        targetAsset = assets.filter((asset: { name: string }) => asset.name.indexOf(`darwin_${arch}`) !== -1)[0];
                         break;
                     case 'linux':
-                        targetAsset = assets.filter((asset: { name: string }) => asset.name.indexOf('linux') !== -1)[0];
+                        targetAsset = assets.filter((asset: { name: string }) => asset.name.indexOf(`linux_${arch}`) !== -1)[0];
                         break;
                     case 'win32':
-                        targetAsset = assets.filter((asset: { name: string }) => asset.name.indexOf('windows') !== -1)[0];
+                        targetAsset = assets.filter((asset: { name: string }) => asset.name.indexOf(`windows_${arch}`) !== -1)[0];
```

Comment:
OPA doesn't publish windows_arm64 binaries, so I think we should skip this here

As for linux and darwin, do we need to add the `_static` suffix too, or it'll find them anyway?

Reply:
_static binaries are detected and installed for me at least. I've tested this branch on an arm Mac and amd64 PC (virtualised) and found it to work correctly.

I think we do need do 'something' when it's a windows arm machine - perhaps just show an error for now which explains there are no suitable binaries?